### ISSUE TITLE
Broken Formatting and Links - 0.12.0~ynh6

### ISSUE THREAD

**Issue by tootbrute** [YunoHost-Apps/writefreely_ynh#88]
### Describe the bug
- blog has no formatting - looks like CSS is broken
- links to posts are broken
- links to tags are broken

### Context
- Hardware: computer at home
- YunoHost version: 4.1.8
- I have access to my server: Through SSH | direct access via keyboard 
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: no
  - If yes, please explain:
- Using, or trying to install package version/branch: 0.12.0~ynh5

### Steps to reproduce
- go to my blog: https://blog.arkadi.one
- click a link

Screenshot 1
https://forum.yunohost.org/uploads/default/original/2X/3/3a30bdf0ad6f5e941037022190797cfe19a513d2.png

Screenshot 2
https://forum.yunohost.org/uploads/default/original/2X/f/fcb17d1ead83e390e43f501c9f16e598b7070687.png

### Expected behavior
- correct formatting
- links work

### Logs
Here is the upgrade to 0.12 log
https://paste.yunohost.org/raw/xamofikaqo

I discussed my problem on the Yunohost Forum too:
https://forum.yunohost.org/t/writefreely-a-minimalist-federated-blog/6804/4

**Comment by yalh76** [comment]
I will look at that, seems like rights issue. Can you check the nginx logas when you try to access your writefreely ?

**Comment by tootbrute** [comment]
> I will look at that, seems like rights issue. Can you check the nginx logas when you try to access your writefreely ?

Not sure if this is the right log...

/var/log/nginx/blog.arkadi.one-error.log.1


```
2021/05/08 20:42:57 [error] 28334#28334: *323733 open() "/var/www/writefreely/static/js/localdate.js" failed (13: Permission denied), client: 88.76.104.41, server: blog.arkadi.one, request: "GET /js/localdate.js HTTP/2.0", host: "blog.arkadi.one", referrer: "https://blog.arkadi.one/photoprism-a-google-photos-replacement"
2021/05/08 20:42:57 [error] 28334#28334: *323733 open() "/usr/share/nginx/html/ynhtheme/custom_overlay.css" failed (2: No such file or directory), client: 88.76.104.41, server: blog.arkadi.one, request: "GET /ynhtheme/custom_overlay.css HTTP/2.0", host: "blog.arkadi.one", referrer: "https://blog.arkadi.one/photoprism-a-google-photos-replacement"
2021/05/08 20:42:57 [error] 28334#28334: *323733 open() "/var/www/writefreely/static/js/webfont.js" failed (13: Permission denied), client: 88.76.104.41, server: blog.arkadi.one, request: "GET /js/webfont.js HTTP/2.0", host: "blog.arkadi.one", referrer: "https://blog.arkadi.one/photoprism-a-google-photos-replacement"
2021/05/08 20:46:16 [error] 28334#28334: *323746 connect() failed (111: Connection refused) while connecting to upstream, client: 95.216.145.65, server: blog.arkadi.one, request: "GET /api/v1/instance HTTP/1.1", upstream: "http://[::1]:8095/api/v1/instance", host: "blog.arkadi.one"
2021/05/08 21:32:10 [error] 28334#28334: *324073 connect() failed (111: Connection refused) while connecting to upstream, client: 54.36.148.220, server: blog.arkadi.one, request: "GET /page/blog.arkadi.one/excession-by-iain-m HTTP/2.0", upstream: "http://[::1]:8095/page/blog.arkadi.one/excession-by-iain-m", host: "blog.arkadi.one"
2021/05/08 22:16:24 [error] 28334#28334: *324498 connect() failed (111: Connection refused) while connecting to upstream, client: 170.39.215.215, server: blog.arkadi.one, request: "GET /api/nodeinfo HTTP/1.1", upstream: "http://[::1]:8095/api/nodeinfo", host: "blog.arkadi.one"
2021/05/08 22:26:42 [error] 28334#28334: *324520 connect() failed (111: Connection refused) while connecting to upstream, client: 116.203.17.124, server: blog.arkadi.one, request: "GET /api/statusnet/config.json HTTP/1.1", upstream: "http://[::1]:8095/api/statusnet/config.json", host: "blog.arkadi.one"
2021/05/08 23:05:30 [error] 28334#28334: *324617 connect() failed (111: Connection refused) while connecting to upstream, client: 114.119.132.153, server: blog.arkadi.one, request: "GET /page/blog.arkadi.one/rocks-and-minerals-by-robert-f HTTP/1.1", upstream: "http://[::1]:8095/page/blog.arkadi.one/rocks-and-minerals-by-robert-f", host: "blog.arkadi.one"
2021/05/08 23:26:06 [error] 28334#28334: *324652 connect() failed (111: Connection refused) while connecting to upstream, client: 114.119.156.131, server: blog.arkadi.one, request: "GET /page/blog.arkadi.one/hurricane-season-by-fernanda-melchor HTTP/1.1", upstream: "http://[::1]:8095/page/blog.arkadi.one/hurricane-season-by-fernanda-melchor", host: "blog.arkadi.one"
2021/05/08 23:50:12 [error] 28334#28334: *324703 connect() failed (111: Connection refused) while connecting to upstream, client: 165.227.199.28, server: blog.arkadi.one, request: "GET /feed/ HTTP/1.1", upstream: "http://[::1]:8095/feed/", host: "blog.arkadi.one"
2021/05/08 23:50:37 [error] 28334#28334: *324717 connect() failed (111: Connection refused) while connecting to upstream, client: 95.216.145.65, server: blog.arkadi.one, request: "GET /api/v1/instance HTTP/1.1", upstream: "http://[::1]:8095/api/v1/instance", host: "blog.arkadi.one"
```

**Comment by wioxjk** [comment]
Links are still broken since this version.
Causing federation and sitemaps to break.

**Comment by tootbrute** [comment]
> Links are still broken since this version.
> Causing federation and sitemaps to break.
> 
> Issue persist when I browser to the app itself via port 8096, so we can exclude nginx
> 
> I personally mitigated the problem by routing the blog via apache and with the following directive:
> 
> ```
> <Location "/blogs.domain.tld/">
>    ProxyPass https://x.x.x.x/
>    ProxyPassReverse https://x.x.x.x/
> </Location>
> ```
> 
> I am sure that similar can be done in NGINX

**Comment by tootbrute** [comment]
Oh shoot. I accidently closed this.

I'm still having this issue with links not working.

WriteFreely uses port 8096. Hmm my Wireguard is now using that port. That's the port Wireguard app automatically chose for me.

**Comment by tootbrute** [comment]
> Links are still broken since this version.
> Causing federation and sitemaps to break.
> 
> Issue persist when I browser to the app itself via port 8096, so we can exclude nginx
> 
> I personally mitigated the problem by routing the blog via apache and with the following directive:
> 
> ```
> <Location "/blogs.domain.tld/">
>    ProxyPass https://x.x.x.x/
>    ProxyPassReverse https://x.x.x.x/
> </Location>
> ```
> 
> I am sure that similar can be done in NGINX

I think I found the file I need to edit to make this happen, but my knowledge of NGINX is abysmal. Everytime I touch NGINX, I breka something.

```
from: /etc/nginx/conf.d/blog.arkadi.one.d/writefreely.conf

location / {
  proxy_pass http://localhost:8095;
  proxy_set_header Host $http_host;
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Forwarded-For $remote_addr;
  proxy_redirect off;

  # Include SSOWAT user panel.
  include conf.d/yunohost_panel.conf.inc;
}
```

Anybody know how to manually fix this?  Doesn't look like a patch is coming.

**Comment by tootbrute** [comment]
Issue with wrong link is now fixed with latest update 0.13.0~ynh1

Thank you so much! I'm closing this issue.

**Comment by wioxjk** [comment]
> location / {
>   proxy_pass http://localhost:8095;
>   proxy_set_header Host $http_host;
>   proxy_set_header X-Real-IP $remote_addr;
>   proxy_set_header X-Forwarded-For $remote_addr;
>   proxy_redirect off;
> 
>   # Include SSOWAT user panel.
>   include conf.d/yunohost_panel.conf.inc;
> }


I think it just like this:

```
location /blogs.domain.tld/ {
  proxy_pass http://localhost:8095;
  proxy_set_header Host $http_host;
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Forwarded-For $remote_addr;
  proxy_redirect off;

  # Include SSOWAT user panel.
  include conf.d/yunohost_panel.conf.inc;
}
```

Just incase someone needs this in the future
